cpu/sam0_common/periph/gpio.c

```diff
@@ -217,15 +232,76 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
     _EIC->WAKEUP.reg |= (1 << exti);
     /* enable the EIC module*/
     _EIC->CTRL.reg = EIC_CTRL_ENABLE;
-    while (_EIC->STATUS.reg & EIC_STATUS_SYNCBUSY) {}
+    EIC_SYNC();
 #else /* CPU_FAM_SAML21 */
     /* enable the EIC module*/
     _EIC->CTRLA.reg = EIC_CTRLA_ENABLE;
-    while (_EIC->SYNCBUSY.reg & EIC_SYNCBUSY_ENABLE) {}
+    EIC_SYNC();
 #endif
     return 0;
 }
 
+inline static void reenable_eic(gpio_eic_clock_t clock) {
+#if defined(CPU_SAMD21)
+    if (clock == _EIC_CLOCK_SLOW) {
+        GCLK->CLKCTRL.reg = EIC_GCLK_ID
+                          | GCLK_CLKCTRL_CLKEN
+                          | GCLK_CLKCTRL_GEN(SAM0_GCLK_32KHZ);
+    } else {
+        GCLK->CLKCTRL.reg = EIC_GCLK_ID
+                          | GCLK_CLKCTRL_CLKEN
+                          | GCLK_CLKCTRL_GEN(SAM0_GCLK_MAIN);
+    }
+    while (GCLK->STATUS.bit.SYNCBUSY) {}
+#else
+    uint32_t ctrla_reg = EIC_CTRLA_ENABLE;
+
+    EIC->CTRLA.reg = 0;
+    EIC_SYNC();
+
+    if (clock == _EIC_CLOCK_SLOW) {
+        ctrla_reg |= EIC_CTRLA_CKSEL;
+    }
+
+    EIC->CTRLA.reg = ctrla_reg;
+    EIC_SYNC();
+#endif
+}
+
+void gpio_pm_cb_enter(int deep)
+{
+#if defined(PM_SLEEPCFG_SLEEPMODE_STANDBY)
+    (void) deep;
+
+    if (PM->SLEEPCFG.bit.SLEEPMODE >= PM_SLEEPCFG_SLEEPMODE_STANDBY) {
```

Comment:
```suggestion
    if (PM->SLEEPCFG.bit.SLEEPMODE == PM_SLEEPCFG_SLEEPMODE_STANDBY) {
```

We can't wake up from HIBERNATE or BACKUP via EXTI anyway, so no need to re-configure the clock.